exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1608 @@
+	}
+	return resp, nil
+}
+
+// UpdateOrderExecutionLimits sets exchange execution order limits for an asset type
+func (d *Deribit) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
+	if !d.SupportsAsset(a) {
+		return fmt.Errorf("%s: %w - %v", d.Name, asset.ErrNotSupported, a)
+	}
+	for _, x := range []string{"BTC", "ETH", "USDC", "USDT"} {
+		var instrumentsData []InstrumentData
+		var err error
+		if d.Websocket.IsConnected() {
+			instrumentsData, err = d.WSRetrieveInstrumentsData(x, d.GetAssetKind(a), false)
+		} else {
+			instrumentsData, err = d.GetInstruments(ctx, x, d.GetAssetKind(a), false)
+		}
+		if err != nil {
+			return err
+		} else if len(instrumentsData) == 0 {
+			continue
+		}
+
+		limits := make([]order.MinMaxLevel, len(instrumentsData))
+		for x := range instrumentsData {
+			var pair currency.Pair
+			pair, err = currency.NewPairFromString(instrumentsData[x].InstrumentName)
+			if err != nil {
+				return err
+			}
+			limits[x] = order.MinMaxLevel{
+				Pair:                   pair,
+				Asset:                  a,
+				PriceStepIncrementSize: instrumentsData[x].TickSize,
+				MinimumBaseAmount:      instrumentsData[x].MinimumTradeAmount,
+			}
+		}
+		err = d.LoadLimits(limits)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetFuturesPositionSummary returns position summary details for an active position
+func (d *Deribit) GetFuturesPositionSummary(ctx context.Context, r *futures.PositionSummaryRequest) (*futures.PositionSummary, error) {
+	if r == nil {
+		return nil, fmt.Errorf("%w HistoricalRatesRequest", common.ErrNilPointer)
+	}
+	if r.Asset != asset.Futures {
+		return nil, fmt.Errorf("%w %v", futures.ErrNotPerpetualFuture, r.Asset)
+	}
+	if r.Pair.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
+	}
+	fPair, err := d.FormatExchangeCurrency(r.Pair, r.Asset)
+	if err != nil {
+		return nil, err
+	}
+	var pos []PositionData
+	if d.Websocket.IsConnected() && d.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+		pos, err = d.WSRetrievePositions(fPair.Base.String(), d.GetAssetKind(r.Asset))
+	} else {
+		pos, err = d.GetPositions(ctx, fPair.Base.String(), d.GetAssetKind(r.Asset))
+	}
+	if err != nil {
+		return nil, err
+	}
+	index := -1
+	for a := range pos {
+		if pos[a].InstrumentName == fPair.String() {
+			index = a
+		}
+	}
+	if index > -1 {
+		return nil, errors.New("position information for the instrument not found")
+	}
+	contracts, err := d.GetFuturesContractDetails(ctx, r.Asset)
+	if err != nil {
+		return nil, err
+	}
+	var multiplier, contractSize float64
+	var settlementType futures.ContractSettlementType
+	for i := range contracts {
+		if !contracts[i].Name.Equal(fPair) {
+			continue
+		}
+		multiplier = contracts[i].Multiplier
+		settlementType = contracts[i].SettlementType
+	}
+
+	var baseSize float64
+	if r.Asset == asset.Futures {
+		baseSize = pos[index].SizeCurrency
+	} else if r.Asset == asset.Options {
+		baseSize = pos[index].Size
```

Comment:
I think you'll have to give this another look:
```
Deribit raw response: {"jsonrpc":"2.0","result":[],"usIn":1713139903110539,"usOut":1713139903110691,"usDiff":152,"testnet":true}
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 11 [running]:
testing.tRunner.func1.2({0x100c3cbe0, 0x1400045a018})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1631 +0x1c4
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1634 +0x33c
panic({0x100c3cbe0?, 0x1400045a018?})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/runtime/panic.go:770 +0x124
github.com/thrasher-corp/gocryptotrader/exchanges/deribit.(*Deribit).GetFuturesPositionSummary(0x1010dac60, {0x100c94600, 0x101142000}, 0x1400063be20)
	/Users/wow/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_wrapper.go:1446 +0x548
github.com/thrasher-corp/gocryptotrader/exchanges/deribit.TestGetFuturesPositionSummary(0x1400009e1a0)
	/Users/wif/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_test.go:2955 +0x1d8
testing.tRunner(0x1400009e1a0, 0x100c8c418)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0xec
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x318

```

You error if the index is > -1, you then look up an index in a slice that must be <= -1 which isn't possible